src/AppInstallerCLICore/Workflows/MultiQueryFlow.cpp

```diff
@@ -0,0 +1,153 @@
+
+            searchContext.Add<Execution::Data::Source>(source);
+            searchContext.Add<Execution::Data::SearchRequest>(std::move(searchRequest));
+
+            packageSubContexts.emplace_back(std::move(searchContextPtr));
+        }
+
+        context.Add<Execution::Data::PackageSubContexts>(std::move(packageSubContexts));
+    }
+
+    void SearchSubContextsForSingle::operator()(Execution::Context& context) const
+    {
+        std::vector<std::unique_ptr<Execution::Context>> packageSubContexts;
+        bool foundAll = true;
+
+        for (auto& searchContextPtr : context.Get<Execution::Data::PackageSubContexts>())
+        {
+            auto& searchContext = *searchContextPtr;
+            SearchRequest searchRequest = searchContext.Get<Execution::Data::SearchRequest>();
+            searchContext.Add<Execution::Data::SearchResult>(searchContext.Get<Execution::Data::Source>().Search(searchRequest));
+
+            switch (m_searchPurpose)
+            {
+            case SearchPurpose::Install:
+                searchContext << Workflow::SelectSinglePackageVersionForInstallOrUpgrade(/* isUpgrade */ false);
+                break;
+            case SearchPurpose::Upgrade:
+                searchContext << Workflow::SelectSinglePackageVersionForInstallOrUpgrade(/* isUpgrade */ true);
+                break;
+            case SearchPurpose::Uninstall:
+                searchContext <<
+                    Workflow::HandleSearchResultFailures <<
+                    Workflow::EnsureOneMatchFromSearchResult(true) <<
+                    Workflow::ReportPackageIdentity;
+                break;
+            default:
+                THROW_HR(E_UNEXPECTED);
+            }
+
+            if (searchContext.IsTerminated())
+            {
+                if (context.IsTerminated() && context.GetTerminationHR() == E_ABORT)
+                {
+                    // This means that the subcontext being terminated is due to an overall abort
+                    context.Reporter.Info() << Resource::String::Cancelled << std::endl;
+                    return;
+                }
+                else
+                {
+                    auto packageString = GetPackageStringFromSearchRequest(searchRequest);
+                    auto searchTerminationHR = searchContext.GetTerminationHR();
+                    if (searchTerminationHR == APPINSTALLER_CLI_ERROR_UPDATE_NOT_APPLICABLE ||
+                        searchTerminationHR == APPINSTALLER_CLI_ERROR_PACKAGE_ALREADY_INSTALLED)
+                    {
+                        AICLI_LOG(CLI, Info, << "Package is already installed: [" << packageString << "]");
+                        context.Reporter.Info() << Resource::String::MultiInstallPackageAlreadyInstalled(packageString) << std::endl;
+                        continue;
+                    }
+                    else
+                    {
+                        AICLI_LOG(CLI, Info, << "Package not found: [" << packageString << "]");

```

Comment:
Looks like it could also be "multiple package found" since we EnsureOneMatch earlier